src/Registry/TypeRegistry.php

```diff
@@ -714,7 +714,7 @@ public function register_type( string $type_name, $config ): void {
 			return $this->prepare_type( $type_name, $config );
 		};
 
-		if ( is_array( $config ) && isset( $config['eagerlyLoadType'] ) && true === $config['eagerlyLoadType'] && ! isset( $this->eager_type_map[ $this->format_key( $type_name ) ] ) ) {
+		if ( \WPGraphQL::is_introspection_query() && is_array( $config ) && isset( $config['eagerlyLoadType'] ) && true === $config['eagerlyLoadType'] && ! isset( $this->eager_type_map[ $this->format_key( $type_name ) ] ) ) {
```

Comment:
Can we `use` the class instead of backlashing it inline?doesn't matter _too_ much in TypeRegistry, but generally:
1. Helps prevent IDEs from getting confused.
2. Makes it easy at a glance at the headers to see the class dependencies.

Not saying we should go through the codebase and update old references, but at least we can prevent new ones from slipping in.